mobile/src/src/components/Community/Feed/FeedContainer.jsx

```diff
@@ -128,13 +128,49 @@ class FeedContainer extends Component {
     return list;
   };
 
-  renderItem({ item }) {
+  attendEvent = (id, shouldAttend) => {
+    let index = this.state.events.findIndex((obj) => obj.id == id);
+    let copyList = this.state.events;
+
+    copyList[index].is_attending = !this.state.events[index].is_attending;
+    this.setState({ copyList: this.state.events });
+
+    EventsEndpoint.attend(
+      id,
+      shouldAttend,
+      () => {},
```

Comment:
All of the lines above EventsEndpoint.attend() should go into this callback function, because we ONLY want to show the updated button if the api call was successful.